addons/web_editor/static/src/xml/ace.xml

```diff
@@ -49,13 +49,34 @@
                 <button data-action="format" type="button" class="btn btn-sm btn-link">Format</button>
             </div>
             <span class="o_ace_editor_resource_info"/>
-            <div class="alert alert-warning alert-dismissible mt-2 mb-0" role="alert">
-                Editing a built-in file through this editor is not advised, as it will prevent it from being updated during future App upgrades.
-                <button type="button" class="btn-close" aria-label="Close"></button>
+        </div>
+        <div id="ace-warning-message" class="mt-2 mb-0" role="alert">
+            <div class="inner">
+                <i class="fa fa-warning"></i>
+                <h1>HTML edits might be lost during future Odoo upgrade.</h1>
+                <p>If you need to add analytics or marketing tags, you can use code injection which is upgrade proof.</p>
+                <i>Note: To embed code in your page, use building block "Custom Code"</i>
+
+                
+                <div class="buttons">
+                    <button type="button" class="btn btn-primary" data-action="edit-header">Inject code in header or body</button>
+                    <button type="button" class="btn btn-secondary" data-action="close">Close editor</button>
+                    <button type="button" class="btn btn-link" data-action="edit-anyway">Edit HTML anyway</button>
+                </div>
             </div>
+            
         </div>
         <div id="ace-view-editor"/>
     </div>
 </t>
 
+<!-- Theme - custom code -->
+<div t-name="web_editor.custom_code_dialog_content">
+    <div class="mb-2" t-esc="headContentText"/>
+    <div class="o_ace_head_editor_container"/>
+    <div class="mb-2" t-esc="bodyContentText"/>
+    <div class="o_ace_body_editor_container"/>
```

Comment:
Could feel overkill but since those two lines are `website.custom_code_dialog_content`, maybe you could move that template in web_editor and t-call it twice here? :+1: